### PR TITLE
send new event if carrier will be switched

### PR DESCRIPTION
**Pull request by AlexSJ1990**

### CODE REVIEW

File: src/main/java/com/hedvig/notificationService/service/event/EventHandler.kt [hunk 1]
```diff
@@ -32,10 +32,16 @@ class EventHandler(
         event: StartDateUpdatedEvent,
         callTime: Instant = Instant.now()
     ) {
+        if (event.carrierWillBeSwitched != null && event.carrierWillBeSwitched) {
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/notificationService/service/event/EventHandler.kt:35]: To always be false for old events that is

File: src/main/java/com/hedvig/notificationService/service/event/EventHandler.kt [hunk 2]
```diff
@@ -32,10 +32,16 @@ class EventHandler(
...
+            logger.info("Handling StartDateWithUpdatedCarrieEvent for ${event.owningMemberId}")
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/notificationService/service/event/EventHandler.kt:36]:
```suggestion
            logger.info("Handling StartDateWithUpdatedCarrierEvent for ${event.owningMemberId}")
```

File: src/main/java/com/hedvig/notificationService/service/event/EventHandler.kt [hunk 3]
```diff
@@ -32,10 +32,16 @@ class EventHandler(
...
+            customerioService.sendEvent(event.owningMemberId, event.toStartDateWithUpdatedCarrieEventMap())
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/notificationService/service/event/EventHandler.kt:37]:
```suggestion
            customerioService.sendEvent(event.owningMemberId, event.toStartDateWithUpdatedCarrierEventMap())
```

File: src/main/java/com/hedvig/notificationService/service/event/EventRequest.kt [hunk 4]
```diff
@@ -119,7 +119,18 @@ data class StartDateUpdatedEvent(
...
     val currentCarrier: Carrier?,
     val carrierOnStartDate: Carrier?
-) : EventRequest()
+) : EventRequest() {
+    fun toStartDateWithUpdatedCarrieEventMap() = mapOf(
+        "name" to "StartDateWithUpdatedCarrieEvent",
+        "data" to mapOf(
+            "member_id" to owningMemberId,
+            "contract_id" to contractId,
+            "carrier_will_be_switched" to carrierWillBeSwitched,
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/notificationService/service/event/EventRequest.kt:128]: The event's existance implies this will always be true

**Review by vonElfvin — APPROVED** [review]

File: src/main/java/com/hedvig/notificationService/service/event/EventRequest.kt (hunk 4 above)
Comment (AlexSJ1990) [reply src/main/java/com/hedvig/notificationService/service/event/EventRequest.kt:128]: gooood point

**Pull request merged**